gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/aggregation/functions/average/AverageProperty.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.aggregation.functions.average;
+
+import org.gradoop.common.model.api.entities.EPGMElement;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.flink.model.impl.operators.aggregation.functions.BaseAggregateFunction;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Base class for aggregate functions determining the average of a numeric property value.
+ */
+public abstract class AverageProperty extends BaseAggregateFunction implements Average {
```

Comment:
Make this not abstract. It can then be used to aggregate edges and vertices at the same time (see MaxProperty). Also add this case to your tests. And remove "base" from the Class/Constructor Javadocs.